src/environment.ts

```diff
@@ -1,8 +1,75 @@
-export default {
-  port: process.env.PORT || 3001,
-  dbHost: process.env.DB_HOST || 'localhost',
-  dbUsername: process.env.DB_USERNAME || 'typescript_user',
-  dbPassword: process.env.DB_PASSWORD || 'password',
-  database: process.env.DATABASE || 'typescript_api',
-  logDB: process.env.LOG_DB !== undefined, // logs all sql commands for gut/fact checking endpoints
+import config from 'config'
+
+type Keys = {
+  privateKey: string
+  publicKey: string
 }
+
+type Certificate = {
+  certificate: string
+  privateKey: string
+}
+
+type Saml = {
+  name: string
+  enabled: boolean
+  attributeMap: Record<string, string>
+  entryPoint: string
+  idpCerts: string[]
+  encryption: Certificate
+  signing: Certificate
+}
+
+type DevAuth = {
+  enabled: boolean
+}
+
+type Providers = {
+  saml: Saml
+  devAuth: DevAuth
+}
+
+function load(path: string): any {
+  if (!config.has(path)) return false
+
+  return config.get(path)
+}
+
+const scheme = (process.env.SCHEME || load('api.scheme') || 'http') as string
+const host = (process.env.HOST || load('api.host') || 'localhost') as string
+const port = (process.env.PORT || load('api.port') || 3001) as number
+
+const apiUrl = `${scheme}://${host}:${port}`
+
+// prettier-ignore
+const refreshTokenExp = process.env.REFRESH_TOKEN_VALIDITY_SECONDS || load('auth.jwt.refreshTokenValiditySeconds') || 864000
+const accessTokenExp = process.env.ACCESS_TOKEN_VALIDITY_SECONDS || load('auth.jwt.accessTokenValiditySeconds') || 600
+const refreshTokenBuffer =
+  process.env.REFRESH_TOKEN_EXPIRATION_BUFFER_SECONDS || load('auth.jwt.refreshTokenExpirationBufferSeconds') || 864000
```

Comment:
Yeah I can go back and forth on these - it would just be nice to be consistent where-ever possible as all of these params are specifically related to JWT stuff